java-frontend/src/main/java/org/sonar/java/reporting/InternalJavaIssueBuilder.java

```diff
@@ -227,32 +232,40 @@ public void report() {
       }
     }
 
-
-    if (!quickFixes.isEmpty() && sonarComponents.isQuickFixCompatible()) {
-      addQuickFixes(ruleKey.get(), (NewSonarLintIssue) newIssue);
+    final List<JavaQuickFix> flatQuickFixes = quickFixes.stream()
+      .flatMap(s -> s.get().stream())
+      .collect(Collectors.toList());
+    if (!flatQuickFixes.isEmpty()) {
+      if (isQuickFixCompatible) {
+        addQuickFixes(inputFile, ruleKey.get(), flatQuickFixes, (NewSonarLintIssue) newIssue);
+      } else if (methodSetQuickFixAvailable != null) {
+        try {
+          methodSetQuickFixAvailable.invoke(newIssue, true);
+        } catch (ReflectiveOperationException e) {
+          throw new RuntimeException(e);
```

Comment:
Throwing this exception here will stop the analysis of the whole file, even if it is unlikely, it seems a bit overkill.